configs/common/packages.yaml

```diff
@@ -59,7 +59,7 @@
     #  version: [8.4.1]
     #  variants: ~xerces ~pnetcdf +parallelio ~shared
     esmf:
-      version: [8.3.0b09]
+      version: [8.4.2]
       variants: ~xerces ~pnetcdf snapshot=b09 ~shared esmf_pio=internal
```

Comment:
These variants need to get updated. `snapshot` should be `none`, and I believe we want to move to `+external-parallelio` (with no `esmf_pio` setting).

Reply:
Also I'm trying to remember the story behind `~shared`. That was for UFS WM, right? Might be good to double check whether we really need it.

Reply:
@AlexanderRichert-NOAA thanks for taking a look. i'll change `snapshot` to `none`. can we get confirmation from @climbfuji regarding removing the `esmf_pio` setting and adding `+external-parallelio`?

i believe `~shared` was included due to those ufs-wm issues we were having a couple months ago. right now, it indeed looks like all the esmf installations from `spack-stack/1.3.0` onward were done with only static libraries. building again w/ shared might bring back those issues and i'd prefer to avoid that as we are close to getting the ufs-wm modulefile pr merged. thoughts?

Reply:
@AlexanderRichert-NOAA esmf variant changes made :) !

Reply:
Cool!